service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseDataExtractionService.java

```diff
@@ -91,104 +92,114 @@ private DocmosisJudgeAndLegalAdvisor getJudgeAndLegalAdvisor(Order standardDirec
-                    .title(formatTitle(direction, standardDirectionOrder.getDirections()))
-                    .body(direction.getDirectionText())
-                    .build())
-                .collect(toList())));
-
-        return formattedDirections.build();
+    // TODO: default value is used here for date of birth, is this correct?
+    private String getDateOfBirth(ChildParty child) {
+        return ofNullable(child.getDateOfBirth())
+            .map(dateOfBirth -> formatLocalDateToString(dateOfBirth, LONG))
+            .orElse(DEFAULT);
     }
 
     private List<DocmosisRespondent> getRespondentsNameAndRelationship(List<Element<Respondent>> respondents) {
         return respondents.stream()
             .map(element -> element.getValue().getParty())
-            .map(respondent -> DocmosisRespondent.builder()
-                .name(respondent.getFullName())
-                .relationshipToChild(defaultIfNull(respondent.getRelationshipToChild(), DEFAULT))
-                .build())
+            .map(this::buildRespondent)
             .collect(toList());
     }
 
-    private List<DocmosisChildren> getChildrenDetails(List<Element<Child>> children) {
-        return children.stream()
-            .map(element -> element.getValue().getParty())
-            .map(child -> DocmosisChildren.builder()
-                .name(child.getFullName())
-                .gender(defaultIfNull(child.getGender(), DEFAULT))
-                .dateOfBirth(getDateOfBirth(child))
-                .build())
-            .collect(toList());
+    // TODO: default value is used for relationship to child, is this correct?
+    private DocmosisRespondent buildRespondent(RespondentParty respondent) {
+        return DocmosisRespondent.builder()
+            .name(respondent.getFullName())
+            .relationshipToChild(defaultIfNull(respondent.getRelationshipToChild(), DEFAULT))
+            .build();
     }
 
-    private String getDateOfBirth(ChildParty child) {
-        return child.getDateOfBirth() == null ? DEFAULT : formatLocalDateToString(child.getDateOfBirth(), LONG);
+    private String getApplicantName(CaseData caseData) {
+        return caseData.findApplicant(0)
+            .map(applicant -> applicant.getParty().getOrganisationName())
+            .orElse("");
     }
 
-    //TODO: look into and potentially refactor -> this seems totally separate to data extraction
-    private String formatTitle(Direction direction, List<DirectionConfiguration> directions) {
-        @AllArgsConstructor
+    private List<DocmosisDirection> getGroupedDirections(CaseData caseData) throws IOException {
+        OrderDefinition standardDirectionOrder = ordersLookupService.getStandardDirectionOrder();
+
+        return ofNullable(caseData.getStandardDirectionOrder().getDirections()).map(directions -> {
+                ImmutableList.Builder<DocmosisDirection> formattedDirections = ImmutableList.builder();
+
+                int index = 2;
```

Comment:
Why do we start with 2? Might be good idea to extract that to const. Also, `index` feels too generic and doesn't provide any information about the variable.

Reply:
yeah this is a weird one, index starts at 2 because direction 1 is hardcoded in sdo. I can extract to constant, this helps if we use this for cmo in future as we can pass in a new starting index. I will change index -> directionNumber